cmd/serverless/main_test.go

```diff
@@ -9,83 +9,47 @@ package main
 
 import (
 	"os"
-	"path"
-	"strings"
+	"syscall"
 	"testing"
+	"time"
 
-	"github.com/stretchr/testify/assert"
-
-	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/serverless/daemon"
 )
 
-func setupTest() {
-	config.Datadog = config.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
-	config.InitConfig(config.Datadog)
-	os.Setenv("AWS_LAMBDA_FUNCTION_NAME", "TestFunction")
-}
-
-func TestProxyNotLoaded(t *testing.T) {
-	setupTest()
-
-	proxyHTTP := "a:1"
-	proxyHTTPS := "a:2"
-	t.Setenv("DD_PROXY_HTTP", proxyHTTP)
-	t.Setenv("DD_PROXY_HTTPS", proxyHTTPS)
-	proxyHTTPConfig := config.Datadog.GetString("proxy.http")
-	proxyHTTPSConfig := config.Datadog.GetString("proxy.https")
-	assert.Equal(t, 0, len(proxyHTTPConfig))
-	assert.Equal(t, 0, len(proxyHTTPSConfig))
-}
-
-func TestProxyLoadedFromEnvVars(t *testing.T) {
-	setupTest()
+func TestDaemonStopOnSIGINT(t *testing.T) {
+	stopCh := make(chan struct{})
+	signalCh := make(chan os.Signal, 1)
 
-	proxyHTTP := "b:1"
-	proxyHTTPS := "b:2"
-	t.Setenv("DD_PROXY_HTTP", proxyHTTP)
-	t.Setenv("DD_PROXY_HTTPS", proxyHTTPS)
+	serverlessDaemon := daemon.StartDaemon("http://localhost:8124")
+	go handleTerminationSignals(serverlessDaemon, stopCh, signalCh)
 
-	config.LoadWithoutSecret()
-	proxyHTTPConfig := config.Datadog.GetString("proxy.http")
-	proxyHTTPSConfig := config.Datadog.GetString("proxy.https")
+	signalCh <- syscall.SIGINT
 
-	assert.Equal(t, proxyHTTP, proxyHTTPConfig)
-	assert.Equal(t, proxyHTTPS, proxyHTTPSConfig)
+	// Use t.Run with a timeout to allow the goroutine to finish
+	t.Run("WaitForStop", func(t *testing.T) {
+		select {
+		case <-stopCh: // Expected behavior, the daemon should be stopped
```

Comment:
A few suggestions for this test:
- SIGINT asnd SIGTERM tests can be combined in a table test reusing all the same code https://golang.cafe/blog/golang-table-test-example
- the test is passing with any signal, it reveals that the `handleTerminationSignals` logic is equivalent to

```golang
signo := <-signalCh
	log.Infof("Received signal '%s', shutting down...", signo)
	serverlessDaemon.Stop()
	stopCh <- struct{}{}
```


To improve the encapsulation of the logic, I would recommend refactoring it in something like
```golang
func handleTerminationSignals(serverlessDaemon *daemon.Daemon, stopCh chan struct{}, notify func(c chan<- os.Signal, sig ...os.Signal)) {
	signalCh := make(chan os.Signal, 1)
	notify(signalCh, syscall.SIGINT, syscall.SIGTERM)
	signo := <-signalCh
	log.Infof("Received signal '%s', shutting down...", signo)
	serverlessDaemon.Stop()
	stopCh <- struct{}{}
}
```
so it can be invoked with
```golang
go handleTerminationSignals(serverlessDaemon, stopCh, signal.Notify)
```

from main and
```golang
go handleTerminationSignals(serverlessDaemon, stopCh, func(c chan<- os.Signal, sig ...os.Signal) {
		c <- syscall.SIGINT
	})
```

- the test is not checking that the `serverlessDaemon` was stopped. It would be great to check the `serverlessDaemon.stopped` flag, or mock it and expect the Stop method to be called